### PR TITLE
Vdom rewrite

### PR DESCRIPTION
**Pull request by Frizi**
Full internal rewrite, exposing the same API.
- issues #95 #79 fixed and covered with regression tests
- 100% test coverage

All inner workings are now using both VDom diffing and Watchers from Vue core. The code should be in general easier to understand.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,46 +1,21 @@
...
-
 const getPath = (ctx, obj, path, fallback) => {
   if (typeof path === 'function') {
     return path.call(ctx, obj, fallback)
   }
 
   path = Array.isArray(path) ? path : path.split('.')
   for (let i = 0; i < path.length; i++) {
-    if (isObject(obj)) {
+    if (typeof obj === 'object') {
```
Comment (kernelgonnapanic) [inline src/index.js]: What about a null check?

Comment (Frizi) [inline src/index.js]: :+1:

**Comment by Frizi** [comment]
> you probably need sb who understands more vuelidate internals

There is nobody yet as it's brand new 😄
Still, concepts are mostly the same, it's the validation VM construction which was rewritten to use Vue's dom diffing.

File: src/index.js [hunk 2]
```diff
@@ -171,185 +47,405 @@ function makePendingAsyncVm (Vue, promise) {
...
-  const validationVm = new Vue({
-    data: {
-      dirty: false,
-      dynamicKeys
+  const ValidationRule = Vue.extend({
+    props: ['rule', 'model', 'parentModel', 'rootModel'],
+    methods: {
+      runRule (parent) {
+        // Avoid using this.parentModel to not get dependent on it.
+        // Passed as an argumend for workaround
```
Comment (kernelgonnapanic) [inline src/index.js]: typo

**Comment by Frizi** [comment]
Released in v0.4.0

**Comment by alexhernandez** [comment]
@Frizi Rewrite seems to be triggering unexpected behaviors with the validators.

**Comment by Frizi** [comment]
@alexhernandez please create an issue and add an example. Hard to say what unexpected behaviour mean without any clues.

**Pull request merged**